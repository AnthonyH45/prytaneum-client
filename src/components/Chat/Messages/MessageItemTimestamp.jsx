import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

const format = (time) => {
    return time;
};

export default function MessageItemTimestamp({ time }) {
    return <Typography variant='body1'>{format(time)}</Typography>;
}

MessageItemTimestamp.propTypes = {
    time: PropTypes.number.isRequired,
};
