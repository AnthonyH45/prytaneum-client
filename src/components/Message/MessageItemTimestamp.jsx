import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { format } from 'date-fns';


export default function MessageItemTimestamp({ time }) {
    return <Typography variant='body1'>{format(new Date(time), 'hh:mm')}</Typography>;
}

MessageItemTimestamp.propTypes = {
    time: PropTypes.number.isRequired,
};
