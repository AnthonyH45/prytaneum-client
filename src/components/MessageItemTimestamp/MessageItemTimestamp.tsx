import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { format } from 'date-fns';

interface Props {
    id?: string;
    time: number;
}

/** Returns a timestamp in the 'hh:mm' using `Date(time)`
 *  @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
 *  @category Component
 *  @constructor MessageItemTimestamp
 *  @param props
 *  @param {string} props.id The id of the wrapping <span>
 *  @param {number} props.time Returns a 'hh:mm' date formulated from time
 */
export default function MessageItemTimestamp({ id, time }: Props) {
    return (
        <span id={id}>
            <Typography variant='body1'>
                {format(new Date(time), 'hh:mm')}
            </Typography>
        </span>
    );
}

MessageItemTimestamp.defaultProps = {
    id: 'MessageItemTimestamp',
};

MessageItemTimestamp.propTypes = {
    time: PropTypes.number.isRequired,
};
