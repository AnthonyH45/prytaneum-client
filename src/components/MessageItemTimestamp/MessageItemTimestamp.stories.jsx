import React from 'react';

import MessageItemTimestamp from '.';

export default { title: 'Components/MessageItemTimestamp' };

const time = 10

export function MessageItemTimestampDisplay() {
    return (
        <div>
            <MessageItemTimestamp time={time} />
            <p>the time=10 should be 04:00</p>
        </div>
    );
}