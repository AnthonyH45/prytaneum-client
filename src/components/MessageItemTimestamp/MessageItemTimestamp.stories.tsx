import React from 'react';
import { format } from 'date-fns';
import MessageItemTimestamp from '.';

export default { title: 'Components/MessageItemTimestamp' };

const time = 10

export function MessageItemTimestampDisplay() {
    return (
        <div>
            <MessageItemTimestamp time={time} />
            <p>the time={time} should be {format(new Date(time), 'hh:mm')}</p>
        </div>
    );
}