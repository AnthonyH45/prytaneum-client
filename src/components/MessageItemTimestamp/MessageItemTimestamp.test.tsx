import React, { Children } from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import { format } from 'date-fns';

import MessageItemTimestamp from './MessageItemTimestamp';

describe('MessageItemTimestamp', function () {
    let container: HTMLDivElement | null = null;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    afterEach(() => {
        // cleanup
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
        }
        container = null;
    });

    it('should render', () => {
        const time = 10;
        const toTest = "<div id=\"MessageItemTimestamp\"><p class=\"MuiTypography-root MuiTypography-body1\">"+format(new Date(time), 'hh:mm')+"</p></div>";
        ReactTestUtils.act(() => {
            render(
                <div id='MessageItemTimestamp'>
                    <MessageItemTimestamp time={time} />
                </div>,
                container
            )
        });
        expect(document.getElementById('MessageItemTimestamp')).toBeTruthy();
        expect(document.getElementById('MessageItemTimestamp')).toContainHTML(toTest);
    });

});
