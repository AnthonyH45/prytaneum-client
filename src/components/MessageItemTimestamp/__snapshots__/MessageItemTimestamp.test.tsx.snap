// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItemTimestamp should render 1`] = `
<span
  id="MessageItemTimestamp"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    04:00
  </p>
</span>
`;
